File: Love Babbar's 450 Coding Question/5.Linked List/Reverse LinkedList.java
// https://practice.geeksforgeeks.org/problems/reverse-a-linked-list/1#
//function Template for Java

/* linked list node class:

class Node {
    int value;
    Node next;
    Node(int value) {
        this.value = value;
    }
}

*/

class Solution
{
    //Function to reverse a linked list.
    Node reverseList(Node head)
    {
        code here
        Node c = head;
        Node p =null; Node n = null;
        
        while(c != null){
            n = c.next;
            c.next = p;
            p = c;
            c = n;
        }
        head = p;
        
        return head;
        
        
    }
}

File: Daily Coding Problem/Bloomberg/reverse_LinkedList.py
# https://leetcode.com/problems/reverse-linked-list/
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        
        c = head
        p = None
        n = None
        
        while c:
            n = c.next
            c.next = p
            p = c 
            c = n
            
        head = p
        
        return head